frontend/src/AddMovie/AddNewMovie/AddNewMovieConnector.js

```diff
@@ -13,13 +15,17 @@ function createMapStateToProps() {
     (state) => state.addMovie,
     (state) => state.movies.items.length,
     (state) => state.router.location,
-    (addMovie, existingMoviesCount, location) => {
+    (state) => state.queue.details.items,
+    createUISettingsSelector(),
+    (addMovie, existingMoviesCount, location, queueItems, uiSettings) => {
       const { params } = parseUrl(location.search);
 
       return {
         ...addMovie,
         term: params.term,
-        hasExistingMovies: existingMoviesCount > 0
+        hasExistingMovies: existingMoviesCount > 0,
+        queueItems,
```

Comment:
Keep the fetch and clear in here, but move the selector part to AddNewMovieSearchResultConnector Since you only use it on a per item basis